### PR TITLE
additional api methods and simplification of the api

### PR DESCRIPTION
**Pull request by dignifiedquire**
depends on https://github.com/n0-computer/iroh/pull/1417

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -48,6 +48,6 @@ tracing-subscriber = { version = "0.3.17" }
 uniffi = { version = "0.24.3", features = ["build"] }
 
 [patch.crates-io]
-iroh = { git = "https://github.com/n0-computer/iroh", branch = "main" }
+iroh = { git = "https://github.com/n0-computer/iroh", branch = "api-improvements" }
```
Comment (b5) [inline Cargo.toml:51]: n0-computer/iroh#1417 has been merged, we can switch this back to `main`

File: src/iroh.udl [hunk 2]
```diff
@@ -32,13 +32,17 @@ interface Doc {
   [Throws=IrohError]
   DocTicket share_read();
   [Throws=IrohError]
-  SignedEntry set_bytes(AuthorId author, bytes key, bytes value);
+  Hash set_bytes(AuthorId author, bytes key, bytes value);
   [Throws=IrohError]
-  bytes get_content_bytes(SignedEntry entry);
+  bytes get_content_bytes(Hash hash);
```
Comment (b5) [inline src/iroh.udl:37]: lovely

File: src/iroh.udl [hunk 3]
```diff
@@ -32,13 +32,17 @@ interface Doc {
...
   [Throws=IrohError]
-  sequence<SignedEntry> latest();
+  sequence<Entry> latest();
   [Throws=IrohError]
   void subscribe(SubscribeCallback cb);
+  [Throws=IrohError]
+  void stop_sync();
```
Comment (b5) [inline src/iroh.udl:43]: I was expecting `stop_sync` to take a document ID argument so I can manage sync on multiple documents. That can totally be a thing for later though. Having this at all is a huge win

File: Cargo.toml (hunk 1 above)
Comment (b5) [reply Cargo.toml]: pushed a commit that updates

**Review by b5 — APPROVED** [review]
LGTM

File: src/iroh.udl (hunk 3 above)
Comment (dignifiedquire) [reply src/iroh.udl:43]: this is a method on the document, so it is already scoped

Comment (b5) [reply src/iroh.udl:43]: OoooOOOOOOOOooOh, I didn't realize. Still learning how to read this `udl` dialect

**Pull request merged**